From 079acf75b485717bebdc6f4710b23b9efe9a2acd Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:11:02 -0400
Subject: [powerpc] modules: comment about de-dotifying symbols when using the ELFv2 ABI

Message-id: <1410545655-205645-434-git-send-email-dzickus@redhat.com>
Patchwork-id: 94340
O-Subject: [RHEL7 PATCH 433/626] powerpc: modules: comment about de-dotifying symbols when using the ELFv2 ABI.
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 5b12c5c69415b184aadb930660a47a8af4c6deb5
Author: Rusty Russell <rusty@rustcorp.com.au>
Date:   Tue Mar 18 20:11:28 2014 +1030

    powerpc: modules: comment about de-dotifying symbols when using the ELFv2 ABI.

    ELFv2 doesn't use function descriptors, so we don't expect symbols to
    start with ".".  But because depmod and modpost strip ".", and we have
    the special symbol ".TOC.", we still need to do it.

    Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/module_64.c b/arch/powerpc/kernel/module_64.c
index 34ba326..05b27a5 100644
--- a/arch/powerpc/kernel/module_64.c
+++ b/arch/powerpc/kernel/module_64.c
@@ -172,6 +172,7 @@ static unsigned long get_stubs_size(const Elf64_Ehdr *hdr,
  return relocs * sizeof(struct ppc64_stub_entry);
 }
 
+/* Still needed for ELFv2, for .TOC. */
 static void dedotify_versions(struct modversion_info *vers,
          unsigned long size)
 {
@@ -182,7 +183,7 @@ static void dedotify_versions(struct modversion_info *vers,
    memmove(vers->name, vers->name+1, strlen(vers->name));
 }
 
-/* Undefined symbols which refer to .funcname, hack to funcname */
+/* Undefined symbols which refer to .funcname, hack to funcname (or .TOC.) */
 static void dedotify(Elf64_Sym *syms, unsigned int numsyms, char *strtab)
 {
  unsigned int i;
-- 
1.7.1